core/src/main/java/org/mskcc/cbio/portal/scripts/ImportProfileData.java

```diff
@@ -92,16 +94,15 @@ public void run() {
             } else if (geneticProfile.getGeneticAlterationType() == GeneticAlterationType.FUSION) {
                 ImportFusionData importer = new ImportFusionData(dataFile, geneticProfile.getGeneticProfileId(), genePanel);
                 importer.importData();
-            } else if (geneticProfile.getGeneticAlterationType() == GeneticAlterationType.GENERIC_ASSAY) {
-
-                // add all missing `genetic_entities` for this assay to the database
-                ImportGenericAssayEntity.importData(dataFile);
-                
-                ImportTabDelimData genericAssayProfileImporter = new ImportTabDelimData(dataFile, geneticProfile.getTargetLine(), geneticProfile.getGeneticProfileId(), genePanel);
-                genericAssayProfileImporter.importData(numLines);
             } else if (geneticProfile.getGeneticAlterationType() == GeneticAlterationType.STRUCTURAL_VARIANT) {
                 ImportStructuralVariantData importer = new ImportStructuralVariantData(dataFile, geneticProfile.getGeneticProfileId(), genePanel);
                 importer.importData();
+            } else if (geneticProfile.getGeneticAlterationType() == GeneticAlterationType.TREATMENT || geneticProfile.getGeneticAlterationType() == GeneticAlterationType.MUTATIONAL_SIGNATURE) {
+                // add all missing `genetic_entities` for this assay to the database
+                ImportGenericAssayEntity.importData(dataFile, geneticProfile.getGeneticAlterationType(), geneticProfile.getOtherMetaDataField("generic_entity_meta_properties"));
+                
+                ImportTabDelimData genericAssayProfileImporter = new ImportTabDelimData(dataFile, geneticProfile.getTargetLine(), geneticProfile.getGeneticProfileId(), genePanel, geneticProfile.getOtherMetaDataField("generic_entity_meta_properties"));
```

Comment:
Do we have to expose "generic_entity_meta_properties" here?  Don't we want them all?

Reply:
@n1zea144 I tried to read the `OtherMetaDataField` from the `DaoGeneticProfile.getGeneticProfileById(geneticProfileId)`. But it's not working since we didn't actually save the `OtherMetaDataField` into the database. So should I keep this?